.github/workflows/make-release-commit.yml

```diff
@@ -36,22 +43,70 @@ jobs:
       uses: actions/setup-python@v4
       with:
         python-version: "3.10"
-    - name: Call bumpversion
-      uses: ./.github/workflows/bump-version
-      with:
-        part: ${{ inputs.part }}
+    
     - name: Create tag
+      id: create_tag
       working-directory: rust
+      env:
+        RELEASE_TYPE: ${{ inputs.type }}
       run: |
-        git add -u
-        git commit -m "Bump version"
-        export TAG="v$(cargo metadata --no-deps --format-version 1 | jq '.packages[0].version' | xargs echo)"
+        # Get the current version from cargo metadata
+        CURRENT_VERSION=$(cargo metadata --no-deps --format-version 1 | jq '.packages[0].version' | xargs echo)
+
+        if [ "$RELEASE_TYPE" == "stable" ]; then
+          echo "Creating stable release for version $CURRENT_VERSION"
+          TAG = "v${CURRENT_VERSION}"
+        else
+          # Get a list of all tags, filter for current version beta tags, sort them and get the last one
+          LAST_BETA_TAG=$(git tag | grep "^v${CURRENT_VERSION}-beta." | sort -V | tail -n 1)
+
+          if [ -z "$LAST_BETA_TAG" ]; then
+            # If there are no existing beta tags for the current version, start with beta.1
+            NEXT_BETA_TAG="v${CURRENT_VERSION}-beta.1"
+          else
+            # If there are existing beta tags, increment the last beta number to get the next one
+            LAST_BETA_NUMBER=$(echo $LAST_BETA_TAG | sed "s/v${CURRENT_VERSION}-beta.//")
+            NEXT_BETA_NUMBER=$((LAST_BETA_NUMBER + 1))
+            NEXT_BETA_TAG="v${CURRENT_VERSION}-beta.${NEXT_BETA_NUMBER}"
+          fi
+
+          echo "Creating beta release for version $CURRENT_VERSION: $NEXT_BETA_TAG"
+          TAG = $NEXT_BETA_TAG
+        fi
+
         git tag $TAG
-    - name: Push new version and tag
+        echo "tag=$TAG" >> $GITHUB_OUTPUT
+    - name: Push new version tag
       if: ${{ inputs.dry_run }} == "false"
       uses: ad-m/github-push-action@master
       with:
         github_token: ${{ secrets.LANCE_RELEASE_TOKEN }}
         branch: ${{ github.ref }}
         tags: true
+    - name: Create GH release
+      if: ${{ inputs.dry_run }} == "false"
+      uses: softprops/action-gh-release@v2
+      with:
+        prerelease: ${{ inputs.type }} == "preview"
+        draft: ${{ inputs.draft-release }} == "true"
+        target_commitish: ${{ steps.create_tag.outputs.tag }}
+        token: ${{ secrets.LANCE_RELEASE_TOKEN }}
+        generate_release_notes: true
```

Comment:
Does this mean we won't manually edit the release notes anymore?

Reply:
We should still give them a look over, but we don't have to click around and create them now.